client/fingerprint/bridge_linux.go

```diff
@@ -43,7 +44,48 @@ func (f *BridgeFingerprint) regexp(pattern, module string) *regexp.Regexp {
 	return regexp.MustCompile(fmt.Sprintf(pattern, module))
 }
 
-func (f *BridgeFingerprint) detect(module string) error {
+func (f BridgeFingerprint) detect(req *FingerprintRequest) error {
+	if err := f.detectKernelModule(bridgeKernelModuleName); err != nil {
+		return fmt.Errorf("failed to detect bridge kernel module: %v", err)
+	}
+
+	return f.detectCNIBinaries(req.Config.CNIPath)
+}
+
+func (f *BridgeFingerprint) detectCNIBinaries(cniPath string) error {
+	if cniPath == "" {
+		return fmt.Errorf("cni is not configured")
+	}
+
+	var errs error
+
+	// plugins required by in client/allocrunner/networking_bridge_linux.go.
+	plugins := []string{"bridge", "firewall", "portmap", "host-local"}
+	for _, plugin := range plugins {
+		if err := f.checkCNIPluginBinary(cniPath, plugin); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func (f *BridgeFingerprint) checkCNIPluginBinary(cniPath, plugin string) error {
+	path := filepath.Join(cniPath, plugin)
+	fi, err := os.Stat(path)
+	if err != nil {
+		return fmt.Errorf("failed to find the %v CNI plugin in %v: %v", plugin, path, err)
+	}
+
+	if !fi.Mode().IsRegular() {
+		return fmt.Errorf("the %v CNI plugin is not a regular file: %v", plugin, path)
+
+	}
```

Comment:
`cni_path` can contain multiple `:` delimited paths, so this needs to use `filepath.SplitList` and loop over the paths.

- docs - https://www.nomadproject.io/docs/configuration/client#cni_path
- code - https://github.com/hashicorp/nomad/blob/v1.1.3/client/allocrunner/networking_cni.go#L75-L76